app/components/ChainSandbox/components/ChainSandboxBody/ChainSandboxBody.tsx

```diff
@@ -55,6 +63,33 @@ const ChainSandboxBody = ({ chainUrl, requestHeaders }: ChainSandboxBodyProps) =
     })
   }, [selectedMethod, isRpc, dispatch])
 
+  const handleCodeEditorChange = useCallback(
+    (value: string) => {
+      trackEvent({
+        category: AnalyticCategories.app,
+        action: AnalyticActions.app_chain_sandbox_send_request,
+        label: `App ID: ${appId}, Blockchain: ${selectedChain?.blockchain}`,
+        value: JSON.stringify(value),
+      })
+      dispatch({ type: "SET_REQUEST_PAYLOAD", payload: value })
+
+      try {
```

Comment:
is this supposed to be inside a useEffect?

Reply:
There's no need for this to be inside a `useEffect`, this part syncs the code change method with the selected method in the "select". The other way around is handled by the event triggered by select, which changes the whole code. 

Also trying to follow best practices mentioned here https://react.dev/learn/you-might-not-need-an-effect